File: hw/jacome/U3/HW17Maze/Maze/src/main/java/ec/edu/espe/maze/controller/MazeController.java
package ec.edu.espe.maze.controller;

import ec.edu.espe.maze.model.Room;
import ec.edu.espe.maze.model.Maze;
import ec.edu.espe.maze.model.MazeGenerator;
import ec.edu.espe.maze.view.MazePrinter;

/**
 *
 * @author Micaela Jácome
 */

/*
Single Responsibility Principle
Dependency Inversion Principle
 */
public class MazeController {

    private final Maze maze;
    private final MazeGenerator generator;
    private final MazePrinter printer;

    public MazeController(int rows, int cols) {
        this.maze = new Maze(rows, cols);
        this.generator = new MazeGenerator(maze);
        this.printer = new MazePrinter();
    }

    public void run() {
        generator.generate();
        maze.getGrid()[0][0].leftWall = false;

        int lastRow = maze.getRows() - 1;
        int lastCol = maze.getCols() - 1;
        maze.getGrid()[lastRow][lastCol].rightWall = false;

        printer.print(maze);
    }

}

File: hw/palacios/u3/HW17Maze/js/controller/MazeController.js
import { Maze } from '../model/Maze.js';
import { MazeGenerator } from '../model/MazeGenerator.js';
import { MazePrinter } from '../view/MazePrinter.js';

export class MazeController {
  constructor(rows, cols) {
    this.maze = new Maze(rows, cols);
    this.generator = new MazeGenerator(this.maze);
    this.printer = new MazePrinter();
  }

  run() {
    this.generator.generate();
    this.maze.getGrid()[0][0].leftWall = false; // Entrada
    const lastRow = this.maze.getRows() - 1;
    const lastCol = this.maze.getCols() - 1;
    this.maze.getGrid()[lastRow][lastCol].rightWall = false; // Salida
    this.printer.print(this.maze);
  }
}